private/rules/jvm_import.bzl

```diff
@@ -15,27 +15,38 @@ def _jvm_import_impl(ctx):
 
     injar = ctx.files.jars[0]
     outjar = ctx.actions.declare_file("processed_" + injar.basename, sibling = injar)
-    ctx.actions.run_shell(
-        inputs = [injar] + ctx.files._add_jar_manifest_entry,
+    args = ctx.actions.args()
+    args.add_all(["--source", injar, "--output", outjar])
+    args.add_all(["--manifest-entry", "Target-Label:{target_label}".format(target_label = ctx.label)])
+    ctx.actions.run(
+        executable = ctx.executable._add_jar_manifest_entry,
+        arguments = [args],
+        inputs = [injar],
         outputs = [outjar],
-        command = "\n".join([
-            # If the jar is signed do not modify the manifest because it will
-            # make the signature invalid.
-            "if unzip -l {injar} | grep -qE 'META-INF/.*\\.SF'; then".format(injar = injar.path),
-            "  cp {injar} {outjar}".format(injar = injar.path, outjar = outjar.path),
-            "else",
-            "  set -e",
-            "  {add_jar_manifest_entry} --source {injar} --manifest-entry 'Target-Label:{target_label}' --output {outjar}".format(
-                add_jar_manifest_entry = ctx.attr._add_jar_manifest_entry.files_to_run.executable.path,
-                injar = injar.path,
-                target_label = ctx.label,
-                outjar = outjar.path,
-            ),
-            "fi",
-        ]),
         mnemonic = "StampJarManifest",
         progress_message = "Stamping the manifest of %s" % ctx.label,
-        tools = [ctx.attr._add_jar_manifest_entry.files_to_run],
+    )
+
+    compilejar = ctx.actions.declare_file("header_" + injar.basename, sibling = injar)
+    args = ctx.actions.args()
+    args.add_all(["--source", outjar, "--output", compilejar])
+    args.add_all(["--remove-entry", "Class-Path"])
+    ctx.actions.run(
+        executable = ctx.executable._add_jar_manifest_entry,
+        arguments = [args],
+        inputs = [outjar],
+        outputs = [compilejar],
+    )
+
+    compilejar = ctx.actions.declare_file("header_" + injar.basename, sibling = injar)
```

Comment:
I'm guessing this wasn't meant to be duplicated?

Reply:
Ooops. Fixed :)